### PR TITLE
WIP: edit project

### PR DESCRIPTION
**Pull request by MasseGuillaume**
* [x] remove foreground/background in favor of https://github.com/scalacenter/scaladex/issues/214

### CODE REVIEW

**Comment by jvican** [comment]
Are you sure you want to set the column limit to 120? It doesn't really make sense to have a scala formatter then... I think that we should follow the 80-char column limit, which is a standard in a lot of projects (like Scalajs).

**Comment by MasseGuillaume** [comment]
@jvican 
The 80-char column limit exist because of the IBM punch card.

I code on a BenQ RL2460HT monitor setup on it's vertical alignment. A 120 char wide column happens to fit perfectly horizontally.

**Comment by jvican** [comment]
Yes, and it keeps being a standard because people read code in mobile phones and small laptops. That's why it's useful to still follow the convention. Also, there are studies showing that vertical code is more readable than horizontal code (take Haskell vs Scala, for instance).

**Comment by heathermiller** [comment]
I agree with Jorge on following convention. I always format to 80-char column widths in everything that I do, and most people I interact with do the same. If 120 is our limit, there's little need for a formatter.

**Comment by MasseGuillaume** [comment]
@Scyks What is your preference for the column size ?

**Comment by heathermiller** [comment]
To be honest, given that we are so weeks past our deadlines, I think wasting time on this argument is rather moot. Instead, maybe we should focus on the goal line?

**Comment by Scyks** [comment]
Actually I prefer a softlimit of 160 chars. In times of big screens there should not be s Problem. Personally I really don't like 80 chars, it makes code more unreadable for me. But I can live with softlimit of 120.

File: model/src/main/scala/ch.epfl.scala.index.model/Project.scala [hunk 1]
```diff
@@ -24,11 +28,18 @@ case class Project(
...
-    customScalaDocUrl: Option[String] = None,
+    artifacts: List[String] = Nil,
+    releaseCount: Int = 0,
+    customScalaDoc: Option[String] = None,
     documentationLinks: Map[String, String] = Map(),
+    deprecated: Boolean = false,
+    contributorsWanted: Boolean = false,
+    artifactDeprecations: Set[String] = Set(),
     logoImageUrl: Option[Url] = None,
+    background: Option[String] = None,
```
Comment (MasseGuillaume) [inline model/src/main/scala/ch.epfl.scala.index.model/Project.scala]: @heathermiller this is actually pretty cool it let project owner customize their page.

File: server/src/main/scala/ch.epfl.scala.index.server/Api.scala [hunk 2]
```diff
@@ -102,21 +100,31 @@ class Api(github: Github)(implicit val ec: ExecutionContext) {
...
                     must(
-                        termQuery("reference.organization", project.organization),
-                        termQuery("reference.repository", project.repository)
+                      references
                     )
                 )
             )
         )
-        .size(5000)
+        .size(500)
```
Comment (MasseGuillaume) [inline server/src/main/scala/ch.epfl.scala.index.server/Api.scala]: artifactReference will filter releases so we only have one artifact (ex: cats-core). This is why I can put a lower limit.

File: server/src/main/scala/ch.epfl.scala.index.server/Server.scala [hunk 3]
```diff
@@ -71,47 +71,48 @@ object Server {
       } yield views.html.frontpage(keywords, targets, dependencies, latestProjects, latestReleases, userInfo)
     }
 
+    def canEdit(owner: String, repo: String, userState: Option[UserState]) = {
+      userState.map(s => s.isAdmin || s.repos.contains(GithubRepo(owner, repo))).getOrElse(false)
+    }
+
+    def editPage(owner: String, repo: String, userState: Option[UserState]) = {
+      val user = userState.map(_.user)
+      // if(canEdit(owner, repo, userState))
```
Comment (MasseGuillaume) [inline server/src/main/scala/ch.epfl.scala.index.server/Server.scala]: Currently working on it. I will re-enable

**Comment by MasseGuillaume** [comment]
a

@heathermiller @Scyks ^

**Comment by MasseGuillaume** [comment]
2016-07-24-204518_2446x1920_scrot

**Pull request merged**